src/containers/createRoom.js

```diff
@@ -9,35 +9,91 @@ export class createRoom extends Component {
     username: "",
   };
 
-  handleChange = (event) => {
+  handleChange(event) {
     this.setState({ [event.target.name]: event.target.value });
   };
 
-  handleSubmit = (event) => {
-    event.preventDefault();
+  getRooms() {
     const reqObj = {
       method: "POST",
       headers: {
         "Content-Type": "application/json",
         Accept: "application/json",
       },
-      body: JSON.stringify({ room: this.state }),
+      body: JSON.stringify({
+        room: this.state
+      }),
     };
-    fetch("http://localhost:3000/rooms", reqObj)
-      .then((resp) => resp.json())
-      .then((resp) => {
-        localStorage.setItem("token", resp.jwt);
-        this.props.setCreateRoom(
-          resp.user,
-          resp.room.room_name,
-          resp.room.host_id,
-          resp.room.host_name
-        );
-        this.props.history.push(`/room/${resp.room.id}`);
-      });
+
+    return fetch("http://localhost:3000/rooms", reqObj)
+      .then((resp) => resp.json());
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    this.getRooms().then((response) => {
+      let { jwt, user, room} = response;
+      
+      localStorage.setItem("token", jwt);
+
+      this.props.setCreateRoom(
+        user,
+        room.room_name,
+        room.host_id,
+        room.host_name
+      );
+      
+      this.props.history.push(`/room/${room.id}`);
+    });
+
     this.setState({ room_name: "", password: "", username: "" });
   };
 
+  renderForm() {
+    return <form className="createRoomForm" onSubmit={this.handleSubmit}>
+      <label className="formLabel">Create Room Name</label>
+      <br></br>
+      <input
+        className="formInput"
+        id="rname"
+        name="room_name"
+        value={this.state.room_name}
+        onChange={this.handleChange}
+      />{" "}
+      <br></br>
+
+      <label className="formLabel">Create Password</label>
+      <br></br>
+      <input
+        className="formInput"
+        id="pword"
+        name="password"
+        type="password"
+        value={this.state.password}
+        onChange={this.handleChange}
+      />
+      <br></br>
+      
+      <label className="formLabel">Create Player Name</label>
```

Comment:
Newlines are your friend. They can visually seperate related lines without the mental overhead of a new function.